### ISSUE TITLE
Lags with minimum of CPU usage

### ISSUE THREAD

**Issue by PhilippTemkin** [UnamSanctam/SilentCryptoMiner#1812]
I created a miner for the hashvault pool, with settings of 40% CPU in normal mode and 80% in idle mode. I installed it on a computer with a 4-core processor, the computer is used only for viewing web pages. Then this miner did not affect the operation of the computer in any way.
After some time, I decided to try another pool, removed the miner from that PC and installed the miner on the 2miners pool with 10% settings in normal mode and disabled the idle mode. I didn’t change the settings, I only changed the pool and processor usage. And the computer began to freeze a lot when browsing websites. That is, in fact, I reduced the load on the processor, but got the opposite effect.

What can be wrong ? Does difficulty on the pool affect these lags? Does connection via SSL/TLS or a simple connection matter? The first miner was connected via an SSL/TLS connection to the pool, the second without a secure connection directly.

**Comment by UnamSanctam** [comment]
Did you make absolutely sure that you removed the old miner? As long as the algorithm was the same then there shouldn't be any difference between computer performance (especially since you chose a lower Max CPU). Is it possible for you to run the checker and post a screenshot of it here?

**Comment by PhilippTemkin** [comment]
> Did you make absolutely sure that you removed the old miner? As long as the algorithm was the same then there shouldn't be any difference between computer performance (especially since you chose a lower Max CPU). Is it possible for you to run the checker and post a screenshot of it here?

Absolutely sure. Because I put a new SSD in this computer and did a clean install of Windows. I can provide a screenshot later/

**Comment by DerXanRam** [comment]
Yeaa i face this problem on a few of my computers that have i5 5th Gen and less CPU

**Comment by DerXanRam** [comment]
especially when chrome opened the PC will stack at miner usage of 20%

**Comment by UnamSanctam** [comment]
>Yeaa i face this problem on a few of my computers that have i5 5th Gen and less CPU
>especially when chrome opened the PC will stack at miner usage of 20%

Then that's most likely because RandomX takes up all the resources (such as cache), only way to avoid that is to mine another algorithm that doesn't use as much resources (each algorithm has a "minimum" amount of resources that has to be used, so nothing can be done in the miner to make it use less than that minimum).

**Comment by PhilippTemkin** [comment]
>Yeaa i face this problem on a few of my computers that have i5 5th Gen and less CPU especially when chrome opened the >PC will stack at miner usage of 20%

Absolutely the same story and the browser is also based on the Chrome kernel. Core i3 9100 processor. Old but still worse than the 5th generation. 4 GB of RAM. This is on that problematic PC. Now I'm writing from a computer on a core i5 10600 with 8GB of RAM and I'm not seeing any problems with the same miner.

**Comment by UnamSanctam** [comment]
>Absolutely the same story and the browser is also based on the Chrome kernel. Core i3 9100 processor. Old but still worse than the 5th generation. 4 GB of RAM. This is on that problematic PC. Now I'm writing from a computer on a core i5 10600 with 8GB of RAM and I'm not seeing any problems with the same miner.

Yes then I would recommend using a different algorithm, for example the RandomX huge pages uses around 2GB of RAM. You can't decrease the amount of RAM needed for huge pages since it's a requirement for the algorithm, you can however disable huge pages but that will decrease the hashrate by a lot. RandomX also uses a lot of other resources like CPU caches.

**Comment by PhilippTemkin** [comment]
> > Yeaa i face this problem on a few of my computers that have i5 5th Gen and less CPU
> > especially when chrome opened the PC will stack at miner usage of 20%
> 
> Then that's most likely because RandomX takes up all the resources (such as cache), only way to avoid that is to mine another algorithm that doesn't use as much resources (each algorithm has a "minimum" amount of resources that has to be used, so nothing can be done in the miner to make it use less than that minimum).

But why did the lags start after the pool change? I didn't change the algorithm. I'm mining Monero, default algorithm is rx/0. this algorithm was in both the first and second. That is, the algorithm itself has not changed.

**Comment by UnamSanctam** [comment]
>But why did the lags start after the pool change? I didn't change the algorithm. I'm mining Monero, default algorithm is rx/0. this algorithm was in both the first and second. That is, the algorithm itself has not changed.

There shouldn't be any difference, if there is one then it's most likely something with the computer. It most likely does not lag because you changed the pool itself but because you stop/start another miner. I could write you a 100 page document for this but the short explanation is most likely one of two things: 1) The miner could not connect to the original pool so it did not mine 2) Now when you start the miner it's able to use more resources than before (like enough RAM for the huger pages), so that manifests in "lag".

**Comment by PhilippTemkin** [comment]
Here is a screenshot of the checker (I censored my personal data) 
![checker](https://github.com/UnamSanctam/SilentCryptoMiner/assets/10112968/833ec92b-c4f9-4f9d-87ac-1890147dddc9)

**Comment by UnamSanctam** [comment]
And you don't change your CPU in the "Remote Configuration"? And have you tried with the newer miner version?

**Comment by PhilippTemkin** [comment]
> And you don't change your CPU in the "Remote Configuration"? And have you tried with the newer miner version?

In the remote configuration there is simply an empty file, I specified this URL in order to be able to reconfigure the miner in the future.

I'm using the old version, but I also tried the same config in the previous version and in the newest version of the miner. But the result is the same, by the way, the size of the miner file itself after compilation in the newest version takes 40+ megabytes, and in version 3.2.0 no more than 6 megabytes.

**Comment by UnamSanctam** [comment]
It should be a smaller size in the new version, are you sure that you don't have "GPU Mining" enabled in the XMR miner in the new version? But if those are the settings then the only way to make it lag less on low-end CPUs would be to use a less demanding algorithm than RandomX.

**Comment by PhilippTemkin** [comment]
I definitely didn't change the config. I didn’t turn on the video card to mine monero (besides, that computer simply doesn’t have one, it’s only on the processor). I’m now far from the computer where the config is stored, but I’m 99.9% sure that I didn’t use the video card.

What algorithms for monero can you recommend? Or where can I find out more about the algorithms supported by monero?

**Comment by UnamSanctam** [comment]
Check if "GPU Mining" is enabled inside the XMR miner "Advanced" in the new version, since that's the only way for it to become 40+MB. Monero uses the RandomX algorithm, so to change algorithm you also need to mine some other cryptocurrency.

**Comment by DerXanRam** [comment]
> Check if "GPU Mining" is enabled inside the XMR miner "Advanced" in the new version, since that's the only way for it to become 40+MB. Monero uses the RandomX algorithm, so to change algorithm you also need to mine some other cryptocurrency.

hello unam ... sorry for my late reply.... can u give me config example for algo that takes low processor and/or RAM? 

Thanks

**Comment by UnamSanctam** [comment]
>hello unam ... sorry for my late reply.... can u give me config example for algo that takes low processor and/or RAM?
Thanks

You could mine some cryptocurrency that uses GhostRider (like RTM), it would probably be easiest to mine it using something like MoneroOcean or Zergpool that can convert what you mine directly into XMR.
